data-mongodb/src/main/java/io/micronaut/data/mongodb/database/MongoReactiveFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.mongodb.database;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.reactivestreams.client.MongoClient;
+import io.micronaut.configuration.mongo.core.DefaultMongoConfiguration;
+import io.micronaut.configuration.mongo.core.NamedMongoConfiguration;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Parameter;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.context.exceptions.ConfigurationException;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.data.mongodb.conf.RequiresReactiveMongo;
+import io.micronaut.data.mongodb.operations.DefaultReactiveMongoRepositoryOperations;
+import io.micronaut.data.mongodb.operations.MongoRepositoryOperations;
+import io.micronaut.data.operations.async.AsyncCapableRepository;
+import io.micronaut.data.operations.reactive.BlockingReactorRepositoryOperations;
+import io.micronaut.data.operations.reactive.ReactorReactiveRepositoryOperations;
+import io.micronaut.data.runtime.operations.ExecutorAsyncOperations;
+import jakarta.inject.Singleton;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * MongoDB reactive beans factory.
+ */
+@RequiresReactiveMongo
+@Internal
+@Factory
+final class MongoReactiveFactory {
+
+    @Primary
+    @Singleton
+    ReactiveMongoDatabaseFactory primaryMongoDatabaseFactory(DefaultMongoConfiguration mongoConfiguration, @Primary MongoClient mongoClient) {
+        return mongoConfiguration.getConnectionString()
+                .map(ConnectionString::getDatabase)
+                .filter(StringUtils::isNotEmpty)
+                .map(databaseName -> new SimpleReactiveMongoDatabaseFactory(mongoClient, databaseName))
+                .orElseThrow(() -> new ConfigurationException("Please specify the default Mongo database in the url string"));
```

Comment:
I think we need make it possible to resolve the database name from the repository definition as well. Just relying in the URL and only allowing one database is a bit inflexible

Reply:
I will add it in another PR